mappings/net/minecraft/recipe/RecipeMatcher.mapping

```diff
@@ -0,0 +1,120 @@
+	METHOD method_7402 match (Lnet/minecraft/class_1860;Lit/unimi/dsi/fastutil/ints/IntList;)Z
+		COMMENT Attempts to match the recipe against the collected inputs.
+		COMMENT Assumes only one output is required.
+		ARG 1 recipe
+			COMMENT The recipe to match against
+		ARG 2 output
+			COMMENT Optional output list of item ids that were matched whilst evaulating the recipe conditions.
+	METHOD method_7403 countCrafts (Lnet/minecraft/class_1860;ILit/unimi/dsi/fastutil/ints/IntList;)I
+		COMMENT Determines the number of crafts that can be produced for a recipe using the
+		COMMENT collected resources available to this crafter.
+		ARG 1 recipe
+			COMMENT The recipe to match against
+		ARG 2 limit
+		ARG 3 output
+			COMMENT Optional output list of item ids that were matched whilst evaulating the recipe conditions.
+	METHOD method_7404 addUnenchantedInput (Lnet/minecraft/class_1799;)V
+		COMMENT Adds a full item stack to the pool of available resources.
+		COMMENT
+		COMMENT <p>This is equivalent to calling {@code addInput(stack, ItemStack.MAX_COUNT)}</p>
+		ARG 1 stack
+	METHOD method_7405 getStackFromId (I)Lnet/minecraft/class_1799;
+		ARG 0 itemId
+	METHOD method_7406 match (Lnet/minecraft/class_1860;Lit/unimi/dsi/fastutil/ints/IntList;I)Z
+		COMMENT Attempts to match the recipe against the collected inputs. Will only succeed if there has been enough
+		COMMENT resources gathered to produce the requested number of outputs.
+		ARG 1 recipe
+			COMMENT The recipe to match against
+		ARG 2 output
+			COMMENT Optional output list of item ids that were matched whilst evaulating the recipe conditions.
+		ARG 3 multiplier
+			COMMENT The number of expected outputs.
+	METHOD method_7407 countCrafts (Lnet/minecraft/class_1860;Lit/unimi/dsi/fastutil/ints/IntList;)I
+		COMMENT Determines the number of crafts that can be produced for a recipe using the
+		COMMENT collected resources available to this crafter.
+		ARG 1 recipe
+			COMMENT The recipe to match against
+		ARG 2 output
+			COMMENT Optional output list of item ids that were matched whilst evaulating the recipe conditions.
+	METHOD method_7408 getItemId (Lnet/minecraft/class_1799;)I
+		ARG 0 stack
+	METHOD method_7409 clear ()V
+	METHOD method_7410 contains (I)Z
+		COMMENT Determines whether a raw item id is present in the pool of crafting resources.
+		ARG 1 itemId
+	METHOD method_7411 consume (II)I
+		COMMENT Consumes a resource from the pool of available items.
+		ARG 1 itemId
+			COMMENT The raw id of the item being consumed.
+		ARG 2 count
+			COMMENT he number of times that item must be consumed.
+	CLASS class_1663 Matcher
+		FIELD field_7551 requiredItems [I
+		FIELD field_7552 ingredients Ljava/util/List;
+		FIELD field_7553 totalRequiredItems I
+		FIELD field_7555 recipe Lnet/minecraft/class_1860;
+		FIELD field_7556 totalIngredients I
+		FIELD field_7557 ingredientItemLookup Lit/unimi/dsi/fastutil/ints/IntList;
+		FIELD field_7558 requirementsMatrix Ljava/util/BitSet;
```

Comment:
This is not a matrix. This is more like just a mapping from int to boolean on whether an ingredient is satisfied etc. Also matrix only makes sense for shaped recipes; it doesn't for shapeless recipes.

Reply:
```suggestion
		FIELD field_7558 requirements Ljava/util/BitSet;
```

Reply:
Hmm, this is actually a 2-d boolean array compressed into a bit set. My bad!